src/app/layout/qgslayoutmanagerdialog.cpp

```diff
@@ -285,6 +292,31 @@ void QgsLayoutManagerDialog::mTemplatesUserDirBtn_pressed()
   openLocalDirectory( mUserTemplatesDir );
 }
 
+void QgsLayoutManagerDialog::createReport()
+{
+  QString title;
+  if ( !QgisApp::instance()->uniqueLayoutTitle( this, title, true ) )
+  {
+    return;
+  }
+
+  if ( title.isEmpty() )
+  {
+    title = QgsProject::instance()->layoutManager()->generateUniqueTitle();
+  }
+
+  std::unique_ptr< QgsReport > report = qgis::make_unique< QgsReport >( QgsProject::instance() );
+  report->setName( title );
+
+  std::unique_ptr< QgsLayout > header = qgis::make_unique< QgsLayout >( QgsProject::instance() );
+  header->initializeDefaults();
+  report->setHeader( header.release() );
+  report->setHeaderEnabled( true );
+
+  QgisApp::instance()->openLayoutDesignerDialog( report.get() );
+  QgsProject::instance()->layoutManager()->addLayout( report.release() );
```

Comment:
I don't see what's the advantage in using a unique_ptr here that we are releasing it and there are no multiple return points neither exceptions, but if this is the modern C++ way, let's follow it!

Reply:
That's what I'm currently leaning toward - even if there's currently no other return points, it doesn't mean the code will always be this way. So in future if someone refactors, they'll have no choice but to respect the ownership of the pointer. It also makes it very clear when reading the code who has ownership - with Qt's parent system it can be difficult when reading over code to know exactly who has ownership of an object, and whether there's a leak or something is correctly parented. Plus, smarter minds then me have declared that modern c++ practice is to *never* use new/delete, and I'm happy to go along with them :) *


* we can't do this always, because of Qt parenting system and also because sip has no concept of smart pointers yet.